Modules/_winapi.c

```diff
@@ -774,12 +774,159 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject*
+sortenvironmentkey(PyObject *module, PyObject *item)
+{
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT,
+                                      LCMAP_UPPERCASE, item);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    "",
+};
+
+static PyObject *
+normalize_environment_keys(PyObject *keys)
+{
+    PyObject *keyfunc = NULL, *kwnames = NULL, *result = NULL;
+    wchar_t *prev_key_string = NULL;
+
+    // Sort the keys.
+    keyfunc = PyCFunction_New(&sortenvironmentkey_def, NULL);
+    if (keyfunc == NULL) {
+        goto error;
+    }
+    kwnames = Py_BuildValue("(s)", "key");
+    if (kwnames == NULL) {
+        goto error;
+    }
+    PyObject *args[] = { keys, keyfunc };
+    if (PyObject_VectorcallMethod(&_Py_ID(sort), args, 1, kwnames) == NULL) {
+        goto error;
+    }
+
+    // Remove duplicated keys and only keep the last inserted one.
+    result = PyList_New(0);
+    if (result == NULL) {
+        goto error;
+    }
+    for (Py_ssize_t i=PyList_GET_SIZE(keys)-1; i>=0; i--) {
+        PyObject *key = PyList_GET_ITEM(keys, i);
+        if (key == NULL) {
```

Comment:
Since `PyList_GET_ITEM` cannot fail, key is never NULL and we can skip the check. At many other places in de codebase `PyList_GET_ITEM` is used without a NULL check. On the other hand, I suspect the check is not performance critical. 

 The `PyList_GET_ITEM` itself is safe because the bounds are checked and the list cannot change size (to check I had to track the calling methods)

Not sure whether there are guidelines for this case.

Reply:
Thanks for the review! I think you are right, updated as you suggested.